pkg/services/accesscontrol/roles.go

```diff
@@ -147,5 +198,23 @@ var PredefinedRoleGrants = map[string][]string{
 	RoleGrafanaAdmin: {
 		usersAdminEdit,
 		usersAdminRead,
+		orgsAdminEdit,
+		orgsAdminRead,
+		ldapAdminEdit,
+		ldapAdminRead,
 	},
+	string(models.ROLE_ADMIN): {
+		orgsCurrentRead,
+		orgsCurrentEdit,
+	},
+}
+
+func concat(permissions []Permission, inheritedPermissions ...[]Permission) []Permission {
+	if inheritedPermissions == nil {
+		return permissions
+	}
+	for _, p := range inheritedPermissions {
+		permissions = append(permissions, p...)
+	}
+	return permissions
 }
```

Comment:
Will have to look this up, but because slices are mutated by `append`, I'm not entirely sure this is safe. It depends on if `permissions` remember its size or if that's mutated as well.

An alternative way of writing this would be something like:

```go
func concat(permissions ...Permission) []Permission {
return permissions
}
```

and call it with

```go
concat(orgsAdminReadRole.Permissions...,
 		Permission{
 			Action: ActionOrgUsersAdd,
 			Scope:  ScopeOrgAllUsersAll,
 		},
 		Permission{
 			Action: ActionOrgUsersRemove,
 			Scope:  ScopeOrgAllUsersAll,
 		},
 		Permission{
 			Action: ActionOrgUsersRoleUpdate,
 			Scope:  ScopeOrgAllUsersAll,
 		},
),
```

Reply:
From [docs](https://golang.org/pkg/builtin/#append)
> The append built-in function appends elements to the end of a slice. If it has sufficient capacity, the destination is resliced to accommodate the new elements. If it does not, a new underlying array will be allocated. Append returns the updated slice. It is therefore necessary to store the result of append, often in the variable holding the slice itself:

Which I read as that this is should be safe, e.g. if there is enough capacity it will append new elements, otherwise allocate a new array.

Reply:
Took an extra look at this: We need a copy, otherwise there's a risk if something is inherited in multiple places, see https://play.golang.org/p/GES977DKhTk for a demo of how that can happen.

Reply:
e30caa76135b1bad0475ec3daee9a24c5a154213 there is a small difference from what you suggested as a method signature, but is is a simple variadic function now, no need to pass two params.